src/scripts/DataLayer.js

```diff
@@ -102,6 +103,7 @@ DataLayer.Manager.prototype._initialize = function() {
  * @private
  */
 DataLayer.Manager.prototype._updateState = function(item) {
+  this._stateCopyBeforeProcessing = this._dataLayer.getState();
```

Comment:
would call this `_previousState`

Reply:
you should just be able to read `this._state` here instead of calling the `dataLayer.getState()` API